### PR TITLE
Avoid fetching logbook data instead in addition to not displaying it

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

#7239 fixed the logbook data being displayed when there was never going to be
any, but I failed to realize that it was still fetching the data and generating 
a database query on the backend 🤦 

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/ha-more-info-logbook.ts [hunk 1]
```diff
@@ -38,48 +38,56 @@ export class MoreInfoLogbook extends LitElement {
...
     if (!stateObj || stateObj.attributes.unit_of_measurement) {
-      return html``;
+      return false;
     }
 
     const domain = computeStateDomain(stateObj);
     if (CONTINUOUS_DOMAINS.includes(domain)) {
-      return html``;
+      return false;
     }
```
Comment (zsarnett) [inline src/dialogs/more-info/ha-more-info-logbook.ts:54]: Move this function below any protected functions

File: src/dialogs/more-info/ha-more-info-logbook.ts [hunk 2]
```diff
@@ -38,48 +38,56 @@ export class MoreInfoLogbook extends LitElement {
...
-        : ""}
-    `;
+    return true;
+  }
+
+  protected render(): TemplateResult {
+    return !this._canHaveLogbookEntries()
+      ? html``
+      : html`
+          ${isComponentLoaded(this.hass, "logbook")
```
Comment (zsarnett) [inline src/dialogs/more-info/ha-more-info-logbook.ts:63]:
```suggestion
  protected render(): TemplateResult {
  	if(!this._canHaveLogbookEntries()) {
  		return html``;
  	}
  	
    return html`
          ${isComponentLoaded(this.hass, "logbook")
```

File: src/dialogs/more-info/ha-more-info-logbook.ts [hunk 3]
```diff
@@ -38,18 +38,26 @@ export class MoreInfoLogbook extends LitElement {
     this._getLogBookData();
   }, 10000);
 
-  protected render(): TemplateResult {
+  private _canHaveLogbookEntries(): boolean {
```
Comment (zsarnett) [inline src/dialogs/more-info/ha-more-info-logbook.ts:41]: move below protected functions

Comment (bdraco) [reply src/dialogs/more-info/ha-more-info-logbook.ts]: done

**Comment by bramkragten** [comment]
We could also prevent the `ha-more-info-logbook` to render in the `more-info` dialog?

**Comment by bdraco** [comment]
I reverted the original change and moved it into `ha-more-info-dialog`

**Review by bramkragten — APPROVED** [review]

**Pull request merged**